recursiveDescentParser.py

```diff
@@ -0,0 +1,122 @@
+from tokens import TokenType, Token
+from nodes import Node, NumberNode, BinaryOpNode, UnaryOpNode, IdentifierNode
+
+
+class Parser():
+    def __init__(self) -> None:
+        self.pos = 0
+        self.tokenList = None
+        self.rootNode: Node = None
+        self.currentToken = None
+        self.stack: dict[Node] = {}
```

Comment:
I love your stack idea! this all feels like magic and you are a wizard!!! HUZZAH 🧙🧙🧙🧙🧙